cpp/ql/src/Security/CWE/CWE-014/MemsetMayBeDeleted.qhelp

```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>Calling <code>memset</code> or <code>bzero</code> on a buffer to clear its contents may get optimized
+away by the compiler if the buffer is not subsequently used. This is not desirable behavior if the buffer
+contains sensitive data that could somehow be retrieved by an attacker.</p>
+
+</overview>
+<recommendation>
+
+<p>Use alternative platform-supplied functions that will not get optimized away. Examples of such
+functions include <code>memset_s</code>, <code>SecureZeroMemory</code>, and <code>bzero_explicit</code>.
```

Comment:
What about `RtlSecureZeroMemory` (mentioned in the old experimental query qhelp)?

Reply:
I'd prefer not to enumerate too many functions on this list. Many Windows functions have kernel versions, and I don't think we should list the kernel versions every time we list the non-kernel function. (But if you think it's important in this case I'll be happy to add it!)

Reply:
That sounds like a good reason to leave it out.